UnityProject/Assets/Scripts/Systems/Clearance/ClearanceRestricted.cs

```diff
@@ -139,5 +103,55 @@ public void SetCheckType(CheckType newType)
 		{
 			type = newType;
 		}
+
+		public static IClearanceSource GrabClearance(GameObject entity)
+		{
+			return GrabClearance(entity, true) as IClearanceSource;
+		}
+
+		public static GameObject GrabClearanceObject(GameObject entity)
+		{
+			return GrabClearance(entity, false) as GameObject;
+		}
+
+		private static object GrabClearance(GameObject entity, bool returnIClearanceSource)
+		{
+			if (entity == null)
+			{
+				return null;
+			}
+			var playerStorage = entity.GetComponent<DynamicItemStorage>();
+			if (playerStorage != null)
+			{
+				List<ItemSlot> slotsToSearch = new List<ItemSlot>();
+				slotsToSearch.Add(playerStorage.GetActiveHandSlot());
```

Comment:
surely best just to grab all slots?

Reply:
Updated it to grab all hand slots instead.